bottlecap/src/lifecycle/invocation/span_inferrer.rs

```diff
@@ -142,15 +146,17 @@ impl SpanInferrer {
         rng.gen()
     }
 
-    /// Returns a reference to the inner `self.inferred_span`
+    /// Returns a clone of the carrier associated with the inferred span
     ///
     #[must_use]
-    pub fn get_inferred_span(&self) -> &Option<Span> {
-        &self.inferred_span
+    pub fn get_carrier(&self) -> Option<HashMap<String, String>> {
```

Comment:
do we need to use getter here instead of .field (maybe with crate pub)?

Reply:
Yes, because we are cloning it instead of passing a reference

Reply:
you can still call span_inferrer.carrier.clone() thought, right?

Reply:
I could make it a public method, but it's recommended to have a getter method when you are doing cloning on non-primitives, all clones are considered heavy in this case